src/main/java/frc/robot/commands/DriveDistance2.java

```diff
@@ -0,0 +1,90 @@
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.Drive;
+import frc.robot.Constants;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class DriveDistance2 extends CommandBase {
+  private final Drive m_subsystem;
+  private double target_distance = 0.0; // desired distance to travel in inches
+  private double error_inches = 2.0; // default error margin in inches
+  private double target_rotations = 0.0; // in sparkmax rotations
+  private double current_rotations = 0.0; // in sparkmax rotations
+  private double error_rotations; //in sparkmax rotations
+
+  // constructor that takes in a distance in inches and uses default error margin
+  public DriveDistance2(Drive subsystem, double distance) {
+    m_subsystem = subsystem;
+    addRequirements(subsystem);
+    target_distance = distance;
+  }
+
+  // constructor that takes in a distance in inches and specified error margin in inches
+  public DriveDistance2(Drive subsystem, double distance, double error) {
+    m_subsystem = subsystem;
+    addRequirements(subsystem);
+    target_distance = distance;
+    if(error < 1.0){
+      error_inches = 1.0;
+    }else{
+      error_inches = error;
+    }
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_subsystem.resetEncoders();
+    current_rotations = 0.0;
+    // convert target distance in inches into sparkmax rotations
+    target_rotations = (inchesToRotations(target_distance));
+    // convert error margin in inches into sparkmax rotations
+    error_rotations = (inchesToRotations(error_inches));
+
+    // should start robot in motion using sparkmax hardware PID/enconders
+    m_subsystem.setLeftRotations(target_rotations);
+    m_subsystem.setRightRotations(target_rotations);
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    // could also add a failsafe to stop after 10 seconds under all circumstances
+    current_rotations =  (m_subsystem.getLeftRotations() + m_subsystem.getRightRotations())/2.0;//takes average, probably unnecssary
+    SmartDashboard.putNumber("Current Inches", rotationsToInches(current_rotations));//display in inches
+    SmartDashboard.putNumber("Target Inches", target_distance);
+    if(current_rotations > (target_rotations - error_rotations) && current_rotations < (target_rotations + error_rotations)){
+      return true;
+    }
+    return false;
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    m_subsystem.resetEncoders();
+    m_subsystem.setLeftRotations(0);
+    m_subsystem.setRightRotations(0);
+    m_subsystem.setDrive(0,0); //probably unnecessary, but just in case
```

Comment:
Generally we shoudl avoid things like "this might be necessary".  Either it is, or it isn't.  And if you have something like this, it basically has to be copied forward forever and noone ever knows if it was necessary or not.

My preference in these scenarios is remove it.  If you have to add it, you add it and you give a good commit message as to why. (or comment here or something) and Then you *know* that it has to be there or doesnt.

Reply:
I guess I meant it as a fail safe. I'd rather stop the robot twice than have it churning into an object. But I'm OK to remove it and understand what you are saying.